docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,114 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be executed from `NEAR CLI`.
+- Extensions are used by a particular group or several groups of users.
+- Extensions are not composable (you should not create extensions for extensions).
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+
+## `Core NEAR CLI` commands
+```txt
+import-account
+    - from-near-wallet
+    - from-seed-phrase <seed-phrase>
+    - from-ledger
+    - from-private-key <private-key>
+
+create-account
+    - implicit
+    - subaccount <master-account> <new-subaccount-id>
```

Comment:
I am not sure how detailed you want to be in this document, but this command will also require the account funding amount in NEAR tokens, the keypair (either specify to generate a new one or ask the user to specify the public key), and at the end, a user would need to sign a transaction (this is in contrast to the implicit account creation flow which does not actually get created on the chain until someone transfers NEAR to it).